source/includes/connection-snippets/x509-snippets/c-connection.c

```diff
@@ -0,0 +1,30 @@
+// begin x509 connection
+#include <mongoc/mongoc.h>
+
+int
+main (int argc, char *argv[])
+{
+   mongoc_database_t *database;
+   mongoc_client_t *client;
+
+   mongoc_init ();
+
+   const mongoc_ssl_opt_t *ssl_default = mongoc_ssl_opt_get_default ();
+   mongoc_ssl_opt_t ssl_opts = { 0 };
+
+   memcpy(&ssl_opts, ssl_default, sizeof ssl_opts);
+
+   ssl_opts.pem_file = "/etc/certs/mongodb/client.pem";
+   ssl_opts.ca_file = "/etc/certs/mongodb/ca.pem";
+   ssl_opts.weak_cert_validation = false;
+   ssl_opts.allow_invalid_hostname = false;
+
+   client = mongoc_client_new ("mongodb+srv://<your-x509-client-subject>@<cluster-url>/?authMechanism=MONGODB-X509");
```

Comment:
Comparing this with other drivers, I think we don't even need to use a `mongoc_ssl_opt_t` struct in this example. The C driver supports the same `tlsCAFile` and `tlsCertificateKeyFile` options used in the Go example. Given that, I think this example file can be drastically reduced and more closely resemble the SCRAM example with the connection string changed.

In fact, every 4.2-compat driver should support those options per the [Unified URI options spec](https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst). You can see the list of compatible drivers by looking at the language tickets linked from https://jira.mongodb.org/browse/DRIVERS-557.

----

@bazile-clyde @kevinAlbs: Since http://mongoc.org/libmongoc/current/authentication.html#x-509-certificate-authentication still demonstrates using `mongoc_ssl_opt_t`, can one of you open a ticket to track updating that to use the new unified options? I think it's sensible to fix it for the 1.15 docs and later, but we could at least do so in the master branch for 1.16+.

Reply:
+1 for using  the URI options instead  of `mongoc_ssl_opt_t`.

Created a ticket to update docs, and to consider deprecating use of that struct entirely:
https://jira.mongodb.org/browse/CDRIVER-3436